PasteBinService/RegisterService/src/main/java/org/potrohit/springboot/RegisterService/infrastructure/s3/S3AdapterImpl.java

```diff
@@ -0,0 +1,139 @@
+package org.potrohit.springboot.RegisterService.infrastructure.s3;
+
+import org.potrohit.springboot.RegisterService.domain.vao.PasteBinVao;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.*;
+import software.amazon.awssdk.services.s3.presigner.S3Presigner;
+import software.amazon.awssdk.services.s3.presigner.model.GetObjectPresignRequest;
+import software.amazon.awssdk.services.s3.presigner.model.PresignedGetObjectRequest;
+import software.amazon.awssdk.services.s3.presigner.model.PresignedPutObjectRequest;
+import software.amazon.awssdk.services.s3.presigner.model.PutObjectPresignRequest;
+
+import java.net.URL;
+import java.time.Duration;
+import java.util.ArrayList;
+
+@Log4j2
+@Component
+public class S3AdapterImpl {
+
+    @Autowired
+    private final S3Client s3Client = null;
+
+    @Autowired
+    private final S3Presigner s3Presigner = null;
+
+    public S3Details getS3Details(final PasteBinVao pasteBinVao) {
+        return S3Details.builder()
+                .s3Bucket("BUCKET")
+                .build();
+    }
+
+    public String getPreSignedDownloadURL(S3Details s3Data) {
+        String s3Bucket = s3Data.getS3Bucket();
+        String s3ObjectKey = s3Data.getS3ObjectKey();
+
+        try {
+            GetObjectRequest objectRequest = GetObjectRequest.builder()
+                    .bucket(s3Bucket)
+                    .key(s3ObjectKey)
+                    .build();
+            GetObjectPresignRequest presignRequest = GetObjectPresignRequest.builder()
+                    .signatureDuration(Duration.ofMinutes(10))
+                    .getObjectRequest(objectRequest)
+                    .build();
+
+            PresignedGetObjectRequest presignedRequest = s3Presigner.presignGetObject(presignRequest);
+            String presignedDownloadURL = presignedRequest.url().toString();
+
+            System.out.println("Presigned URL to download a file to: " +presignedDownloadURL);
+            System.out.println("Which HTTP method needs to be used when uploading a file: " + presignedRequest.httpRequest().method());
+            return presignedDownloadURL;
```

Comment:
multiple returns are not encouraged. Each method should have single point of entry and exit. So define a variable about try call it presignedurl and assign it to null. 
In the try assign the value. And return it in 62.